### DISCUSSION TITLE
SQLITE_BUSY: database is locked

### DISCUSSION THREAD

**Discussion by prodigerati** [outline/outline#1742] [Self Hosting]
I'm having a couple of issues with my install, I hope someone can provide some assistance.

The first is images stop loading from AWS after a period of time. The only thing that resolves the issue is a server reboot.
I have turned on DEBUG and noticed the following in the error logs:

`0|outline  |   retry: { max: 5, match: [ 'SQLITE_BUSY: database is locked' ] },`

The second issue I have is not being able to update my logo in the [settings -> details] section of Outline. I am able to select a logo file and it displays correctly in the crop preview but once I hit save it shows a ? ... 

![Screen Shot 2020-12-23 at 10 36 25 AM](https://user-images.githubusercontent.com/631360/103027980-bcd58a00-450b-11eb-9f52-13f657429663.png)

Im running on Digital Ocean Ubuntu VM with PM2, Nginx proxy, ssl, slack, google, AWS. Let me know if I can provide any additional logs or config that will be helpful.

**Comment by tommoor** [comment]
Hey, this log is strange as Outline does not use or support SQLite – perhaps that's your issue?

**Comment by prodigerati** [comment]
Installed Postgres Version - psql (12.5 (Ubuntu 12.5-0ubuntu0.20.04.1))

Sqlite not installed:
`sudo apt-get autoremove sqlite3`
`Reading package lists... Done`
`Building dependency tree`
`Reading state information... Done`
`Package 'sqlite3' is not installed, so not removed`
`0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.`

`sudo apt-get autoremove sqlite`
`Reading package lists... Done`
`Building dependency tree`
`Reading state information... Done`
`Package 'sqlite' is not installed, so not removed`
`0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.`

**Comment by prodigerati** [comment]
By clicking refresh in the browser twice the images do return without a reboot. I can live with that.

Any ideas on how to get a logo uploaded and working?

I tried to set a direct url in the .env file and it shows as ? on both the login screen and after logging in.
`TEAM_LOGO=https://aws-url/logo.png`

**Reply by tommoor** [reply]
The error will be in your browser logs, most likely the ACL permissions in S3 do not allow it to be loaded without a signed url or the signed url you are using has expired. I'd recommend setting to public access for this usecase

**Reply by prodigerati** [reply]
I did eventually figure this out, thanks for the information. AWS can be tricky for permissions.
